src/psyclone/tests/dynamo0p3_build.py

```diff
@@ -32,34 +32,101 @@
+                              "argument_mod",
+                              "kernel_mod",
+                              "partition_mod",
+                              "mesh_map_mod",
+                              "mesh_mod",
+                              "stencil_dofmap_mod",
+                              "function_space_mod",
+                              "field_mod",
+                              "abstract_quadrature_mod",
+                              "quadrature_rule_mod",
+                              "quadrature_xyz_mod",
+                              "quadrature_xyoz_mod",
+                              "quadrature_xoyoz_mod",
+                              "quadrature_mod",
+                              "operator_mod"]
+
+    def __init__(self, tmpdir):
+        '''Constructor for the Dynamo0p3-specific compilation class.
+        The very first time the constructor is called it will compile
+        the infrastructure library in a temporary, process-specific
+        location. These files will be used by all test compilations.
+        :param tmpdir: Temporary directory to be used for output files.
+        :type tmpdir: :py:class:`LocalPath`
+        '''
+        super(Dynamo0p3Build, self).__init__(tmpdir,)
+
+        import os
+        base_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 "test_files", "dynamo0p3")
+        self.base_path = base_path
+        self._infrastructure_path = os.path.join(base_path, "infrastructure")
+        # On first instantiation (triggered by conftest.infra_compile)
+        # compile the infrastructure library files.
+        if not Dynamo0p3Build._infrastructure_built:
+            self._build_infrastructure()
+
+    def get_infrastructure_flags(self):
+        '''Returns the required flag to use the infrastructure wrapper
+        files for dynamo0p3. Each parameter must be a separate entry
+        in the list, e.g.: ["-I", "/some/path"] and not ["-I /some/path"].
+        :returns: A list of strings with the compiler flags required.
+        :rtpe: list
+        '''
+        return ["-I", Dynamo0p3Build._compilation_path]
+
+    def _build_infrastructure(self):
+        '''Compiles the Dynamo0.3 wrapper infrastructure files so that
+        compilation tests can be done.
+        '''
+        old_pwd = self._tmpdir.chdir()
+        # Store the temporary path so that the compiled infrastructure
+        # files can be used by all test compilations later.
+        Dynamo0p3Build._compilation_path = str(self._tmpdir)
+
+        try:
+            # Compile each infrastructure file
+            for fort_file in Dynamo0p3Build.INFRASTRUCTURE_MODULES:
+                name = self.find_fortran_file([self._infrastructure_path],
+                                              fort_file)
+                # Abort if there should be a problem
+                if not self.compile_file(name):
+                    break
```

Comment:
`Dynamo0p3Build._infrastructure_built` will still get set to `True` if this break statement is executed. Should we raise an Exception of some sort instead?